packages/vite/src/node/optimizer/esbuildDepPlugin.ts

```diff
@@ -190,30 +194,31 @@ export function esbuildDepPlugin(
         }
       })
 
-      build.onLoad(
-        { filter: /.*/, namespace: 'browser-external' },
-        ({ path: id }) => {
-          return {
-            contents:
-              `export default new Proxy({}, {
-  get() {
-    throw new Error('Module "${id}" has been externalized for ` +
-              `browser compatibility and cannot be accessed in client code.')
-  }
-})`
-          }
-        }
-      )
-
       // yarn 2 pnp compat
       if (isRunningWithYarnPnp) {
         build.onResolve(
           { filter: /.*/ },
-          async ({ path, importer, kind, resolveDir }) => ({
-            // pass along resolveDir for entries
-            path: await resolve(path, importer, kind, resolveDir)
-          })
+          async ({ path: id, importer, kind, resolveDir, namespace }) => {
+            const resolved = await resolve(
+              id,
+              importer,
+              kind,
+              // pass along resolveDir for entries
+              namespace === 'dep' ? resolveDir : undefined
+            )
+            if (resolved) {
+              return resolveResult(id, resolved)
+            }
+          }
         )
+
+        // Without pnp esbuild is transforming modules disabled for the browser
+        // to empty CommonJS modules. In pnp mode we achieve the same by passing
+        // empty contents causing esbuild to produce an empty CommonJS module.
+        build.onLoad({ filter: /.*/, namespace: 'browser-external' }, () => {
+          return { contents: '' }
+        })
```

Comment:
Okay, seems this https://github.com/vitejs/vite/pull/6493#discussion_r803042275 was an incorrect assumption.

Test is failing in `playground/optimize-deps/dep-linked-include`.

`dep-linked-include` is written as a node-based package. It has no `browser` mapping, only `main` linking to `index.mjs` which has a direct import from `fs`.

So instead of be handled by ESBuild itself, it goes through esbuildDepPlugin.

Interestingly, it differs from a situation when a builtin is imported in a source code directly rather than in a pre-bundled dependency such as dep-linked-include. As these are handled by `load` from resolvePlugine whereas dep-linked-include is handled by esbuildDepPlugin's onLoad.

Seemingly similar cases go through different code paths.

Perhaps it'd be better if Vite just instructed ESBuild to bundle this package for the browser instead of overriding resolution 🤔